text/0060-tasks-api.md

```diff
@@ -0,0 +1,578 @@
+- Title: Tasks API
+- Start Date: 2021-08-13
+- Specification PR: [#60](https://github.com/meilisearch/specifications/pull/60)
+- Discovery Issue: [#48](https://github.com/meilisearch/product/issues/48)
+
+# Refashion Updates APIs
+
+## 1. Functional Specification
+
+### I. Summary
+
+The term `update` is not the best choice as it can be confused with document updates or `settings`. We have chosen to replace this term with `task`, which is more generic and better reflects the meaning of this API resource.
+
+As an additional change, we have reworked the format of an update to make it more in line with our expectations of an API that is supposed to be easily understandable and developer-oriented.
+
+Changing the format of `task` object lists makes adding more functionality in a future iteration easier. See the Future Possibilities section for a brief overview.
+
+The specification adds two new API endpoints. Although quite simple, they allow consulting the list of tasks or a specific task without being forced to know the related index.
+
+The specification makes any writing operation on an index asynchronous to serve consistency and facilitate future evolutions.
+
+#### Summary Key Points
+
+- The `update` resource is renamed `task`. The names of existing API routes are also changed to reflect this change.
+- Tasks are now also accessible as an independent resource of an index. `GET - /tasks`; `GET - /tasks/:taskUid`
+- The task `uid` is not incremented by index anymore. The sequence is generated globally.
+- A `task_not_found` error is introduced.
+- The format of the `task` object is updated.
+    - `updateId` becomes `uid`.
+    - Attributes of an error appearing in a `failed` `task` are now contained in a dedicated `error` object.
+    - `type` is no longer an object. It now becomes a string containing the values of its `name` field previously defined in the `type` object.
+    - The possible values for the `type` field are reworked to be more clear and consistent with our naming rules.
+    - A `details` object is added to contain specific information related to a `task` payload that was previously displayed in the `type` nested object.
+    - An `indexUid` field is added to give information about the related index on which the task is performed.
+    - `duration` format has been updated to express an `ISO 8601` duration.
+    - `processed` status changes to `succeeded`.
+    - `startedProcessingAt` is updated to `startedAt`.
+    - `processedAt` is updated to `finishedAt`.
+- `202 Accepted` requests previously returning an `updateId` are now returning a summarized `task` object.
+- `MEILI_MAX_UDB_SIZE` env var is updated `MEILI_MAX_TASK_DB_SIZE`.
```

Comment:
Hi @brunoocasali!  

In general, we do this by specification in dedicated pull-requests while they are in reviews but this is a bit special, it's the pull-request that gathers the specifications that will go into main when v0.25 is released... So it's never too late to fix some typos! Thanks 😃

Indeed it is a replacement because we have renamed the resource `update` by `task`. In this case, `U` meant update but `UDB` as such was not explicit enough.